src/core/router/info.cc

```diff
@@ -723,21 +723,31 @@ void RouterInfo::CreateBuffer(const PrivateKeys& private_keys)
       if (!m_Buffer)
         m_Buffer = std::make_unique<std::uint8_t[]>(Size::MaxBuffer);
       std::memcpy(m_Buffer.get(), router_info.Str().c_str(), m_BufferLen);
+    }
+  catch (...)
+    {
+      m_Exception.Dispatch(__func__);
+      throw;
+    }
+}
 
-      // Signature
-      // TODO(anonimal): signing should be done when creating RI, not after. Requires other refactoring.
-      private_keys.Sign(
-          reinterpret_cast<std::uint8_t*>(m_Buffer.get()),
-          m_BufferLen,
-          reinterpret_cast<std::uint8_t*>(m_Buffer.get()) + m_BufferLen);
-
-      m_BufferLen += private_keys.GetPublic().GetSignatureLen();
+bool RouterInfo::Verify()
+{
+  bool success = false;
+  try
+    {
+      std::size_t len = m_BufferLen - m_RouterIdentity.GetSignatureLen();
+      if (len < Size::MinUnsignedBuffer)
+        throw std::length_error("RouterInfo: invalid RouterInfo size");
+      success =
```

Comment:
The bool'ing isn't necessary, the function should return void. We'll either throw or not depending on outcome of verifier's bool.

Reply:
Ok, wasn't sure if I should carry on the pattern. Will change with your recommendation.